astropy/stats/sigma_clipping.py

```diff
@@ -54,7 +55,11 @@ def _nanmean(array, axis=None):
     if isinstance(axis, tuple):
         array = _move_tuple_axes_first(array, axis=axis)
         axis = 0
-    return bottleneck.nanmean(array, axis=axis)
+
+    if isinstance(array, Quantity):
+        return bottleneck.nanmean(array, axis=axis) * array.unit
```

Comment:
This is a perfect candidate for the new `<<` operator! (No unnecessary copy)

Arguably better would be to also do `array.view(np.ndarray)` since now things will break if `bottleneck` becomes subclass-sensitive. Though, actually, `<<` prevents that too...

Note, though, that this does not keep, say, `Angle`...

As yet another alternative, which works for `ndarray` and `Quantity` and should also work for any well-written `ndarray` subclass:
```
return array.__array_wrap__(bottleneck.nanmean(array))
```

Reply:
I changed this to use `__array_wrap__`, thanks!